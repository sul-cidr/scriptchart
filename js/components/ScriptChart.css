table, th, td {
  color: black;
  border: 1px solid gray;
}

td img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 50px;
  height: 50px;
}

th {
  padding: 5px;
  text-align: center;
}

td {
  padding: 5px;
  text-align: left;
}

td:hover {
  background-color: #bbbbbb;
}

tr:nth-child(even) {
  background-color: #f2f2f2
}