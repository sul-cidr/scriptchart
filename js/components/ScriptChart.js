import React from "react";

/* ScriptChart - This is the central script chart/table rendering
 * component, situated between its parent DashTabs component, which
 * stores most of the state of the scriptchart contents, and the
 * DragTable component, which implements lower-level functions of
 * the scriptchart.
 * It instantiates the rows and columns via props passed down from
 * submission of the selection/configuration form (columnManuscripts,
 * rowLetters), and populates them -- filling in the rows -- 
 * with data queried from the API.
 * It also passes up requests to move and hide rows and columns
 * and to display manuscripts in the viewer, and is re-rendered
 * when these changes are processed.
 */

import DragTable from "./DragTable";
import LetterImage from "./LetterImage";
import ColumnControls from "./ColumnControls";
import SyriacLetter from "./SyriacLetter";

import "./index.css";

class ScriptChart extends React.Component {
  constructor(props) {
    super(props);

    this.getRows = this.getRows.bind(this);
    this.getColumns = this.getColumns.bind(this);
    this.onRow = this.onRow.bind(this);
    this.onHideColumn = this.onHideColumn.bind(this);
    this.onHideRow = this.onHideRow.bind(this);
    this.viewManifest = this.viewManifest.bind(this);
  }

  /* Table data is stored as a two-dimensional array.
   * First key: letter ID (row)
   * Second key: manuscript ID (column)
   * Value: an Object containing the data necessary to render
   * 1-5 image letter excerpts (binarized or not). This includes
   * the link to the page URL and the coordinates of each letter.
   */

  viewManifest(manifestURL) {
    this.props.onManifestSelected(manifestURL);
  }

  onHideColumn(manuscriptID, columnHider) {
    columnHider(manuscriptID);
  }

  getRows() {
    let rows = [];

    /* Add row that will contain link to Mirador viewer, column hider Xs */
    let colControls = { id: 0, ltid: "", letter: "", visible: true };

    /* Add dates row */
    let datesRow = {
      id: 1,
      ltid: "Date",
      letter: "Date",
      visible: !this.props.hiddenLetters.includes("Date")
    };

    for (let i = 0, len = this.props.columnManuscripts.length; i < len; i++) {
      colControls["manuscript" + (i + 1)] = (
        <ColumnControls
          msid={this.props.columnManuscripts[i].id}
          shelfmark={this.props.columnManuscripts[i].shelfmark}
          manifestURL={this.props.columnManuscripts[i].manifest}
          displayManifest={this.viewManifest}
          onHideColumn={this.onHideColumn}
          onHiddenChange={this.props.onHiddenChange}
        />
      );
      datesRow["manuscript" + (i + 1)] = this.props.columnManuscripts[i].date;
    }
    rows.push(colControls);
    rows.push(datesRow);

    /* Load the letters data into the rows array */
    for (let i = 0, len = this.props.rowLetters.length; i < len; i++) {

      let ltID = this.props.rowLetters[i].id;
      let row = {
        id: i + 2,
        ltid: ltID,
        letter: <SyriacLetter key={ltID} id={ltID} />,
        visible: !this.props.hiddenLetters.includes(ltID)
      };

      /* This is where the actual letter instances from the manuscript
       * pages are added */
      for (let j = 0, len = this.props.columnManuscripts.length; j < len; j++) {
        let msID = this.props.columnManuscripts[j].id;

        if (
          !(msID in this.props.tableData) ||
          !(ltID in this.props.tableData[msID])
        ) {
          row["manuscript" + (j + 1)] = <div msid={msID} />;
        } else {
          let cellContents = this.props.tableData[msID][ltID]
            .slice(0, this.props.formData.letterExamples)
            .map(coords => {
              return (
                <LetterImage
                  key={coords.id}
                  letter={this.props.rowLetters[i].letter}
                  coords={coords}
                  imageSize={this.props.formData.imageSize}
                  imageDisplay={this.props.formData.imageDisplay}
                  cropMargin={this.props.formData.cropMargin}
                />
              );
            });
          row["manuscript" + (j + 1)] = <div msid={msID}>{cellContents}</div>;
        }
      }

      rows.push(row);
    }
    return rows;
  }

  getColumns() {
    /* First two columns are the images of the letters and the "X" to hide
     * each row.
     * XXX The transparent header labels are being used to wedge the column
     * widths to a uniform setting -- need to find a better way to do this.
     */
    let cols = [
      {
        property: "letter",
        header: {
          label: <span style={{ color: "transparent" }}>Letter</span>
        },
        props: {
          style: { width: 80 }
        }
      }
    ];
    let rowRemoverColumn = {
      property: "row_remover",
      header: {
        label: <span style={{ color: "transparent" }}>X</span>
      },
      props: {
        style: { width: 45 }
      },
      cell: {
        formatters: [
          (value, { rowData }) =>
            rowData.id > 0 ? (
              <span
                title="Hidden row"
                className="remove"
                onClick={() => this.onHideRow(rowData.ltid)}
                style={{ cursor: "pointer" }}
              >
                &#10007;
              </span>
            ) : (
              <span />
            )
        ]
      },
      visible: true
    };
    cols.push(rowRemoverColumn);
    /* Iteratively populate the columns */
    for (let i = 0, len = this.props.columnManuscripts.length; i < len; i++) {
      // XXX Consider just not generating values for hidden columns...
      let column = {
        property: "manuscript" + (i + 1),
        header: {
          label: this.props.columnManuscripts[i].shelfmark,
          props: {
            label: this.props.columnManuscripts[i].shelfmark,
            style: { width: 200 },
            onMove: o => this.props.onColumnMove(o)
          }
        },
        visible: !this.props.hiddenManuscripts.includes(
          this.props.columnManuscripts[i].id
        ),
        props: {
          msid: this.props.columnManuscripts[i].id,
          style: { width: 200 }
        }
      };
      cols.push(column);
    }
    return cols;
  }

  onRow(row) {
    return {
      rowId: row.ltid,
      onMove: this.props.onRowMove
    };
  }

  onHideRow(letterID) {
    this.props.onHiddenChange("hide", "row", letterID);
  }

  render() {
    console.log("Rendering ScriptChart");

    let rows = this.getRows();
    let columns = this.getColumns();

    return (
      <DragTable
        rows={rows}
        columns={columns}
        hiddenLetters={this.props.hiddenLetters}
        hiddenManuscripts={this.props.hiddenManuscripts}
        onRow={this.onRow}
      />
    );
  }
}

export default ScriptChart;
