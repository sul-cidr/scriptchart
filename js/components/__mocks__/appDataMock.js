export const tableData = {"9":{"2":[{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Alaph (Angular)_611_277_75_50.png","id":44407,"top":277,"left":611,"width":75,"height":50},{"page":613,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-31a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_31a_Alaph (Angular)_557_289_70_48.png","id":44412,"top":289,"left":557,"width":70,"height":48},{"page":1481,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-28a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_28a_Alaph (Angular)_1016_249_64_44.png","id":44408,"top":249,"left":1016,"width":64,"height":44},{"page":1481,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-28a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_28a_Alaph (Angular)_727_357_63_44.png","id":44409,"top":357,"left":727,"width":63,"height":44},{"page":2902,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-11b.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_11b_Alaph (Angular)_252_393_55_46.png","id":44402,"top":393,"left":252,"width":55,"height":46}],"19":[{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Kaph_304_277_44_39.png","id":44503,"top":277,"left":304,"width":44,"height":39},{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Kaph_727_342_41_40.png","id":44504,"top":342,"left":727,"width":41,"height":40},{"page":613,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-31a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_31a_Kaph_1034_350_43_44.png","id":44507,"top":350,"left":1034,"width":43,"height":44},{"page":613,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-31a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_31a_Kaph_959_463_42_38.png","id":44508,"top":463,"left":959,"width":42,"height":38},{"page":1156,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-13b.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_13b_Kaph_504_762_42_38.png","id":44501,"top":762,"left":504,"width":42,"height":38}],"37":[{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Sadhe_732_1071_106_59.png","id":44593,"top":1071,"left":732,"width":106,"height":59},{"page":49,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-22b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_22b_Sadhe_465_1427_75_54.png","id":44594,"top":1427,"left":465,"width":75,"height":54},{"page":1156,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Vat.%20Syriac%20112,%20Ephrem,%20Hymns%20on%20Paradise%20(551_%2093%20fols.)/MSS_Vat.sir.112-13b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_13b_Sadhe_582_1301_76_51.png","id":44592,"top":1301,"left":582,"width":76,"height":51},{"page":1377,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr112/VatSyr112-006.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_3L_Sadhe_735_855_61_46.png","id":9863,"top":855,"left":735,"width":61,"height":46},{"page":1377,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr112/VatSyr112-006.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 112_3L_Sadhe_636_1134_62_50.png","id":9864,"top":1134,"left":636,"width":62,"height":50}]},"11":{"2":[{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_201_258_23_24.png","id":74759,"top":258,"left":201,"width":23,"height":24},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_278_254_22_24.png","id":74760,"top":254,"left":278,"width":22,"height":24},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_125_281_28_25.png","id":74761,"top":281,"left":125,"width":28,"height":25},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_212_283_22_20.png","id":74762,"top":283,"left":212,"width":22,"height":20},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Alaph (Angular)_237_278_22_25.png","id":74763,"top":278,"left":237,"width":22,"height":25}],"19":[{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Kaph_675_615_15_15.png","id":13256,"top":615,"left":675,"width":15,"height":15},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Kaph_299_214_18_13.png","id":13294,"top":214,"left":299,"width":18,"height":13},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Kaph_608_731_16_15.png","id":13346,"top":731,"left":608,"width":16,"height":15},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Kaph_289_518_18_13.png","id":10180,"top":518,"left":289,"width":18,"height":13},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Kaph_585_808_15_13.png","id":10181,"top":808,"left":585,"width":15,"height":13}],"37":[{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_432_234_21_21.png","id":13238,"top":234,"left":432,"width":21,"height":21},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_420_262_24_19.png","id":13239,"top":262,"left":420,"width":24,"height":19},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_624_417_23_21.png","id":13248,"top":417,"left":624,"width":23,"height":21},{"page":7,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-291.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_145L_Sadhe_647_296_26_21.png","id":13259,"top":296,"left":647,"width":26,"height":21},{"page":208,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/VatSyr140/VatSyr140-003.png","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/Vat. Syr. 140_1R_Sadhe_760_293_26_17.png","id":10253,"top":293,"left":760,"width":26,"height":17}]},"16":{"2":[{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_759_1756_86_99.png","id":70463,"top":1756,"left":759,"width":86,"height":99},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_636_1925_95_93.png","id":70464,"top":1925,"left":636,"width":95,"height":93},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_667_1133_68_75.png","id":70465,"top":1133,"left":667,"width":68,"height":75},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_1377_1609_72_77.png","id":70466,"top":1609,"left":1377,"width":72,"height":77},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":2,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Alaph (Angular)_1721_1791_86_82.png","id":70467,"top":1791,"left":1721,"width":86,"height":82}],"19":[{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_976_1150_46_63.png","id":70558,"top":1150,"left":976,"width":46,"height":63},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1083_1970_42_55.png","id":70559,"top":1970,"left":1083,"width":42,"height":55},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1334_2351_46_61.png","id":70560,"top":2351,"left":1334,"width":46,"height":61},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1734_1993_43_57.png","id":70561,"top":1993,"left":1734,"width":43,"height":57},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":19,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Kaph_1354_1708_50_54.png","id":70562,"top":1708,"left":1354,"width":50,"height":54}],"37":[{"page":440,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-46b.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_46b_Sadhe_843_1144_72_78.png","id":32009,"top":1144,"left":843,"width":72,"height":78},{"page":930,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-57a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_57a_Sadhe_1667_2193_83_86.png","id":70663,"top":2193,"left":1667,"width":83,"height":86},{"page":1799,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-49a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_49a_Sadhe_713_1940_92_95.png","id":32007,"top":1940,"left":713,"width":92,"height":95},{"page":2147,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-46a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_46a_Sadhe_1105_1224_82_82.png","id":70659,"top":1224,"left":1105,"width":82,"height":82},{"page":2147,"pageurl":"https://images.syriac.reclaim.hosting/manuscripts/Add.17224/add17724-46a.jpg","letter":37,"binaryurl":"https://images.syriac.reclaim.hosting/binary/BL. Add. 17224_46a_Sadhe_1616_1408_82_84.png","id":70660,"top":1408,"left":1616,"width":82,"height":84}]}};

export const formData = {"showBinarized":true,"letterExamples":3,"imageSize":"Small","selectedShelfmarks":["Vat. Syr. 112","Vat. Syr. 140","BL. Add. 17224"],"manuscripts":[],"letters":[{"id":2,"letter":"Alaph (Angular)","is_script":true,"display":"ܐ","script":"estrangela"},{"id":25,"letter":"Lamadh (Final, open)","is_script":true,"display":"ܠ","script":"estrangela"},{"id":37,"letter":"Sadhe","is_script":true,"display":"ܨ","script":"estrangela"}]};

export const columns = [{"property":"letter","visible":true,"props":{"style":{"width":80}}},{"property":"row_remover","props":{"style":{"width":45}},"cell":{"formatters":[null]},"visible":true},{"property":"manuscript1","header":{"label":"Vat. Syr. 112","props":{"label":"Vat. Syr. 112"}},"visible":true,"props":{"msid":9,"style":{"width":200}}},{"property":"manuscript2","header":{"label":"Vat. Syr. 140","props":{"label":"Vat. Syr. 140"}},"visible":true,"props":{"msid":11,"style":{"width":200}}},{"property":"manuscript3","header":{"label":"BL. Add. 17224","props":{"label":"BL. Add. 17224"}},"visible":true,"props":{"msid":16,"style":{"width":200}}}];