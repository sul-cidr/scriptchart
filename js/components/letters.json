[
  {
    "id": 2,
    "letter": "Alaph (Angular)",
    "is_script": true,
    "display": "ܐ",
    "script": "estrangela"
  },
  {
    "id": 4,
    "letter": "Alaph (Round)",
    "is_script": true,
    "display": "ܐ",
    "script": "serto"
  },
  {
    "id": 9,
    "letter": "Beth",
    "is_script": true,
    "display": "ܒ",
    "script": "estrangela"
  },
  {
    "id": 14,
    "letter": "Gamal",
    "is_script": true,
    "display": "ܓ",
    "script": "estrangela"
  },
  {
    "id": 12,
    "letter": "Dalath (Angular)",
    "is_script": true,
    "display": "ܕ",
    "script": "estrangela"
  },
  {
    "id": 13,
    "letter": "Dalath (Round)",
    "is_script": true,
    "display": "ܕ",
    "script": "serto"
  },
  {
    "id": 16,
    "letter": "He (Angular)",
    "is_script": true,
    "display": "ܗ",
    "script": "estrangela"
  },
  {
    "id": 17,
    "letter": "He (Round)",
    "is_script": true,
    "display": "ܗ",
    "script": "serto"
  },
  {
    "id": 46,
    "letter": "Waw",
    "is_script": true,
    "display": "ܘ",
    "script": "estrangela"
  },
  {
    "id": 50,
    "letter": "Zain",
    "is_script": true,
    "display": "ܙ",
    "script": "estrangela"
  },
  {
    "id": 18,
    "letter": "Heth",
    "is_script": true,
    "display": "ܚ",
    "script": "estrangela"
  },
  {
    "id": 45,
    "letter": "Teth",
    "is_script": true,
    "display": "ܛ",
    "script": "estrangela"
  },
  {
    "id": 48,
    "letter": "Yudh (Connected)",
    "is_script": true,
    "display": "ܝ",
    "script": "estrangela",
    "leading_letter": "ܝ",
    "trailing_letter": "ܝ"
  },
  {
    "id": 49,
    "letter": "Yudh (Stand-alone)",
    "is_script": true,
    "display": "ܝ",
    "script": "estrangela"
  },
  {
    "id": 19,
    "letter": "Kaph",
    "is_script": true,
    "display": "ܟ",
    "script": "estrangela",
    "leading_letter": "ܝ"
  },
  {
    "id": 20,
    "letter": "Kaph (Final)",
    "is_script": true,
    "display": "ܟ",
    "script": "estrangela"
  },
  {
    "id": 25,
    "letter": "Lamadh (Final, open)",
    "is_script": true,
    "display": "ܠ",
    "script": "estrangela"
  },
  {
    "id": 21,
    "letter": "Lamadh",
    "is_script": true,
    "display": "ܠ",
    "script": "estrangela",
    "trailing_letter": "ܝ"
  },
  {
    "id": 23,
    "letter": "Lamadh (Final, closed)",
    "is_script": true,
    "display": "ܠ",
    "script": "serto"
  },
  {
    "id": 26,
    "letter": "Mim",
    "is_script": true,
    "display": "ܡ",
    "script": "estrangela",
    "leading_letter": "ܝ"
  },
  {
    "id": 27,
    "letter": "Mim (Final)",
    "is_script": true,
    "display": "ܡ",
    "script": "estrangela"
  },
  {
    "id": 28,
    "letter": "Nun",
    "is_script": true,
    "display": "ܢ",
    "script": "estrangela",
    "leading_letter": "ܝ"
  },
  {
    "id": 30,
    "letter": "Nun (Final, connected)",
    "is_script": true,
    "display": "ܢ",
    "script": "estrangela",
    "trailing_letter": "ܝ"
  },
  {
    "id": 31,
    "letter": "Nun (Final, unconnected)",
    "is_script": true,
    "display": "ܢ",
    "script": "estrangela"
  },
  {
    "id": 38,
    "letter": "Semkath",
    "is_script": true,
    "display": "ܣ",
    "script": "estrangela"
  },
  {
    "id": 5,
    "letter": "Ayin",
    "is_script": true,
    "display": "ܥ",
    "script": "estrangela"
  },
  {
    "id": 32,
    "letter": "Pe",
    "is_script": true,
    "display": "ܦ",
    "script": "estrangela"
  },
  {
    "id": 37,
    "letter": "Sadhe",
    "is_script": true,
    "display": "ܨ",
    "script": "estrangela"
  },
  {
    "id": 33,
    "letter": "Qaph",
    "is_script": true,
    "display": "ܩ",
    "script": "estrangela"
  },
  {
    "id": 35,
    "letter": "Rish (Angular)",
    "is_script": true,
    "display": "ܪ",
    "script": "estrangela"
  },
  {
    "id": 36,
    "letter": "Rish (Round)",
    "is_script": true,
    "display": "ܪ",
    "script": "serto"
  },
  {
    "id": 39,
    "letter": "Shin",
    "is_script": true,
    "display": "ܫ",
    "script": "estrangela"
  },
  {
    "id": 43,
    "letter": "Taw (Looped)",
    "is_script": true,
    "display": "ܬ",
    "script": "estrangela"
  },
  {
    "id": 44,
    "letter": "Taw (Triangular)",
    "is_script": true,
    "display": "ܬ",
    "script": "serto",
    "trailing_letter": "ܝ"
  },
  {
    "id": 42,
    "letter": "Taw (L-shaped)",
    "is_script": true,
    "display": "ܬ",
    "script": "serto"
  }
]
