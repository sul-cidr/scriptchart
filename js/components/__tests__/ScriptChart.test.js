/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import ScriptChart from "../ScriptChart";
import HTML5Backend from "react-dnd-html5-backend";
import { DragDropContext } from "react-dnd";
import letters from "../letters.json";
import { tableData, defaultManuscripts } from "../__mocks__/appDataMock";

jest.mock("../DragTable");

const DDContext = DragDropContext(HTML5Backend)(ScriptChart);

describe("ScriptChart test", () => {
  const formData = {
    showBinarized: true,
    letterExamples: 3,
    imageSize: "Small",
    selectedShelfmarks: ["Vat. Syr. 112", "Vat. Syr. 140", "BL. Add. 17224"],
    manuscripts: [],
    letters: [
      {
        id: 2,
        letter: "Alaph (Angular)",
        is_script: true,
        display: "ܐ",
        script: "estrangela"
      },
      {
        id: 25,
        letter: "Lamadh (Final, open)",
        is_script: true,
        display: "ܠ",
        script: "estrangela"
      },
      {
        id: 37,
        letter: "Sadhe",
        is_script: true,
        display: "ܨ",
        script: "estrangela"
      }
    ]
  };
  it("ScriptChart should match snapshot", () => {
    const component = renderer.create(
      <DDContext
        hiddenLetters={letters.slice(0, 3)}
        rowLetters={letters.slice(0, 5)}
        columnManuscripts={defaultManuscripts.slice(0, 4)}
        hiddenManuscripts={defaultManuscripts.slice(0, 3)}
        tableData={tableData}
        formData={formData}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
