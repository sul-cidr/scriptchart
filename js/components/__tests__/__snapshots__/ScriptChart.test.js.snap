// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptChart test ScriptChart should match snapshot 1`] = `null`;
